### PR TITLE
Backout SE-0246

### PR DESCRIPTION
**Pull request by stephentyrone**
We're working on figuring out a better source- and binary-stability solution for this change (hopefully with a good back-deployment solution as well). Since we didn't end up taking it into Swift-5.1, I'm going to back it out of master in the short-term while we replan.

Resolves <rdar://problem/53956572>

### CODE REVIEW

**Comment by stephentyrone** [comment]
@swift-ci please test

File: test/api-digester/Outputs/stability-stdlib-abi.without.asserts.swift.expected [hunk 1]
```diff
@@ -0,0 +1,133 @@
+Func Double.acos(_:) has been removed
```
Comment (lorentey) [inline test/api-digester/Outputs/stability-stdlib-abi.without.asserts.swift.expected]: This looks as if we have the digester baseline set to a recent master here. That seems wrong to me -- @nkcsgexi, shouldn't we use the 5.0 (or 5.1) state?

**Review by lorentey — COMMENTED** [review]
Looks good!

File: test/api-digester/Outputs/stability-stdlib-abi.without.asserts.swift.expected (hunk 1 above)
Comment (stephentyrone) [reply test/api-digester/Outputs/stability-stdlib-abi.without.asserts.swift.expected]: Yup, I agree (was in the process of writing up a radar about that =))

Comment (stephentyrone) [reply test/api-digester/Outputs/stability-stdlib-abi.without.asserts.swift.expected]: <rdar://problem/54653088>

**Comment by compnerd** [comment]
This may have caused a regression on the android x64 build: https://dev.azure.com/compnerd/windows-swift/_build/results?buildId=7070&view=logs&j=4805d3d2-0b91-5df3-5741-d0ea1cc955c0

**Comment by stephentyrone** [comment]
@compnerd yeah, it looks like the removal of `Float80` on Android builds happened during the period when 246 was on master, so it needs to be applied to BuiltinMath as well.

**Comment by stephentyrone** [comment]
@compnerd https://github.com/apple/swift/pull/26824

**Comment by rxwei** [comment]
FYI: We are keeping `ElementaryFunctions` in the `tensorflow` branch since TensorFlow optimizers depend on this.

**Pull request merged**